File: src/lambda/dms-switch/utils/get-dms-config.ts
import { CloudFormationClient } from '@aws-sdk/client-cloudformation';
import { listStackResources } from '../../dms-switch/utils/list-stack-resources';

export const getDmsConfig = async ({ cf, StackName }: {
  cf: CloudFormationClient;
  StackName: string;
}): Promise<string> => {
  const resources = await listStackResources({ cf, StackName, resources: [] });
  const dmsConfig = resources.filter((res) => res.ResourceType === "AWS::DMS::ReplicationConfig")[0].PhysicalResourceId;
  console.log(dmsConfig);
  return `${dmsConfig}`;
};

File: src/lambda/dms-switch-py/utils/get_dms_config.py
from utils.list_stack_resources import list_stack_resources


def get_dms_config(cf, stack_name):
    resources = list_stack_resources(cf, stack_name, [])
    dms_configs = [res['PhysicalResourceId']
                   for res in resources if res['ResourceType'] == 'AWS::DMS::ReplicationConfig']
    if dms_configs:
        dms_config = dms_configs[0]
        print(dms_config)
        return dms_config
    else:
        return None
